### PR TITLE
feat(TDP-3389): SSO authentication integration

### PR DESCRIPTION
**Pull request by frassinier**
**Link to the JIRA issue**
e.g. https://jira.talendforge.org/browse/TDP-3389

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/933/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/933/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/934/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/934/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/935/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/935/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/936/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/936/)

File: dataprep-webapp/src/app/index-config.js [hunk 1]
```diff
@@ -26,7 +29,22 @@ function getAppConfig() {
 
 function getAppSettings(config) {
 	const url = `${config.serverUrl}/api/settings`;
-	return getConfig(url);
+	const middleware = ($http) => {
+		const loginUrl = config.loginUrl;
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/index-config.js]: loginUrl should be added to config.json?

File: dataprep-webapp/src/app/index-config.js [hunk 2]
```diff
@@ -26,7 +29,22 @@ function getAppConfig() {
...
+		if (loginUrl) {
+			$http.defaults.transformResponse = [
+				// Same role as a HTTP interceptor
+				(data, headersGetter, status) => {
+					if (status === 401) {
+						window.location.href = loginUrl;
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/index-config.js]: $window instead?

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1019/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1019/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1055/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1055/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1056/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1056/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1057/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1057/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1059/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1059/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1073/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1073/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1117/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1117/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1118/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1118/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1119/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1119/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1126/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1126/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1128/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1128/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/11/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/11/)

File: dataprep-webapp/src/app/index-config.js (hunk 2 above)
Comment (frassinier) [inline dataprep-webapp/src/app/index-config.js]: Angular services are not yet loaded

File: dataprep-webapp/src/app/index-config.js (hunk 1 above)
Comment (frassinier) [inline dataprep-webapp/src/app/index-config.js]: :+1: yes this only concerns EE

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/30/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/30/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/34/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/34/)

**Pull request merged**